File: fixtures/lang-coverage/csharp/Service.cs
// feature: class inheriting BaseService, implementing IGreeter,
//          using directives (all styles), cross-file call, external call
using System;
using System.Collections.Generic;
using static System.Console;

namespace Graphy
{
    public class Service : BaseService, IGreeter
    {
        private Dictionary<string, string> _cache;

        public Service(string name) : base(name)
        {
            _cache = new Dictionary<string, string>();
        }

        public string Hi()
        {
            return "hello from " + Name;
        }

        public void Run()
        {
            string greeting = Helpers.FormatName(Name);
            Console.WriteLine(greeting);
            void LocalLog(string msg) { Console.Write(msg); }
            LocalLog(greeting);
        }
    }
}

File: fixtures/lang-coverage/elixir/lib/service.ex
defmodule Service do
  # feature: defmodule, alias, import, require, use, def, call
  alias Helpers
  import String, only: [upcase: 1]
  require Logger

  def run(name) do
    greeting = Helpers.format_name(name)
    greeting
  end

  def describe(name) do
    "Service(#{name})"
  end

  defp private_run(name) do
    run(name)
  end
end
